src/windows/wslaservice/exe/WSLAVirtualMachine.cpp

```diff
@@ -587,13 +601,21 @@ std::tuple<int32_t, int32_t, wsl::shared::SocketChannel> WSLAVirtualMachine::For
     return std::make_tuple(pid, ptyMaster, wsl::shared::SocketChannel{std::move(socket), std::to_string(pid), m_vmTerminatingEvent.get()});
 }
 
-wil::unique_socket WSLAVirtualMachine::ConnectSocket(wsl::shared::SocketChannel& Channel, int32_t Fd)
+std::pair<int, wil::unique_socket> WSLAVirtualMachine::ConnectSocket(wsl::shared::SocketChannel& Channel, int32_t Fd)
```

Comment:
minor, i think a simple struct would be better than a pair. I think it would make the code more reable if you could say socket.Fd and socket.Handle versus socket.first and socket.second

Reply:
Agreed, done in latest iteration.